src/Refactoring2-Transformations/RBInsertClassTransformation.class.st

```diff
@@ -72,21 +72,27 @@ RBInsertClassTransformation >> addClass: aName superclass: superclassName subcla
 
 { #category : #preconditions }
 RBInsertClassTransformation >> preconditions [
+	"Superclass and subclasses shouldn't be metaclasses.
+	Each subclass should be immediate subclass of superclass."
 
 	| cond |
-	cond := ((RBCondition isMetaclass: (self model classObjectFor: superclass))
-				errorMacro: 'Superclass must not be a metaclass') not.
-	cond := subclasses inject: cond into: [:sub :each |
-		sub & ((RBCondition isMetaclass: (self model classObjectFor: each))
-					errorMacro: 'Subclass must <1?not :>be a metaclass') not
-			 & (RBCondition
-					isImmediateSubclass: (self model classObjectFor: each)
-					of: (self model classObjectFor: superclass))].
+	cond := ((RBCondition isMetaclass:
+		          (self model classObjectFor: superclass)) errorMacro:
+		         'Superclass must not be a metaclass') not.
+	cond := subclasses inject: cond into: [ :sub :each |
+		        sub
+		        &
+			        ((RBCondition isMetaclass:
+				          (self model classObjectFor: each)) errorMacro:
+				         'Subclass must <1?not :>be a metaclass') not
+		        & (RBCondition
+				         isImmediateSubclass: (self model classObjectFor: each)
+				         of: (self model classObjectFor: superclass)) ].
 	^ cond & (RBCondition isValidClassName: className)
-		& (RBCondition isGlobal: className in: self model) not
-			& (RBCondition isSymbol: category)
-			& ((RBCondition withBlock: [category isEmpty not])
-					errorMacro: 'Invalid category name')
+	  & (RBCondition isGlobal: className in: self model) not
+	  & (RBCondition isSymbol: category)
+	  & ((RBCondition withBlock: [ category isNotEmpty ]) errorMacro:
```

Comment:
Not for this PR (as this was already like that), but & is not that good as it evaluates all the cases and it is not optimized, while #and: is optimized and does not evaluate the block if the sender is false.

Reply:
Ah, this is the & on RBCondition... so my comment is wrong